src/components/Home/News/NewsManager.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState } from "react"
+import { DeviceEventEmitter } from "react-native"
+import AsyncStorage from "@react-native-async-storage/async-storage"
+
+import { api } from "api"
+import { Article, Tag } from "api/articles"
+import {
+    TagWithData,
+    Preference,
+    Preferences,
+    LastArticles,
+    UpdatePreferenceEventArgs,
+    UPDATE_PREFERENCE_EVENT_NAME,
+} from "./newsTypes"
+import { NewsBottomSheet } from "./NewsBottomSheet"
+import { newsTagsPatterns, CardsPattern } from "utils/cardsPatterns"
+
+/**
+ * Bottom sheet in the home page to access the news.
+ *
+ * This is the component that retrieves and precesses the news content.
+ * Afterwards, it passes the data to the NewsBottomSheet.
+ */
+export const NewsManager = () => {
+    const [tags, setTags] = useState<Tag[]>([])
+
+    // Object to store the preference of each tag
+    const [preferences, setPreferences] = useState<Preferences>({})
```

Comment:
allora sta cosa è ottima, giusto che sia contenuto in uno state. la cosa che non mi va un granché giù è l'utilizzo degli event emitter per tenere traccia dello stato di preferenza.

essendo che la cosa ha senso che sia completamente stateful (anche perché il tuo approccio è già quasi completamente stateful, i.e usi gli useEffect per i side effect perfettamente) avrebbe più senso secondo me che anche questo behaviour sia governato dagli stessi principi.

prova a vedere se ti torna più comodo fare qualcosa tipo passare questo state come parametri di navigazione (che spero funzioni e si aggiornino con lo state tipo props, non ho provato), con cose tipo `onSetPreference` o qualcosa del genere per avere un bubbling di questo state nel tree di navigazione.
altrimenti puoi sempre considerare la possibilità di scrivere un piccolo context consumato da tutte ste pagine ("piccolo" nel senso non globale, necessariamente nel MainStack perché deve contenere tutte le pagine in cui lo vuoi usare) per condividere questo state, se preferisci così puoi prendere ispirazione con altri context tipo Settings e Login

Reply:
Non so se ho capito bene, ma se intendi passare una funzione che aggiorna le preferenze al posto di usare l'event emitter, avevo provato così all'inizio. Solo che passare dei parametri non serializzabili (funzioni) come parametri di navigazione genera uno warning, e anche se l'utilizzo dell'event emitter non piace nemmeno a me, non mi era venuto in mente nient'altro. Quindi vada per il piccolo context?

Reply:
ah shit ok ahah, allora sì context è l'ideale